src/animate.py

```diff
@@ -24,28 +25,61 @@
 matplotlib.rc('text', usetex='false')
 
 
-def run(country, to_plot='confirmed', save=False, path=None):
+def run(country, region= "", to_plot='confirmed', save=False, path=None,cached=False,repeat=True):
     """Create the animation.
 
     Parameters
     ----------
     country: str
         Country to plot.
+    region: str
+        Region of the given contry, currently only for Italy supportorted ('all', NAME or mNAME for excluding NAME region)
     to_plot: str, default 'confirmed'
         'confirmed' for confirmed cases, or 'deaths' for confirmed deaths.
     save: bool, default False
         Whether to save the plot.
     path: str, default None
         Path where to save the plot (if save == False). If None, save it in 'examples/'
+    cached: bool, default False
+        Whether to cache the data source in a local file
+    repeat: bool, default True
+        Whether to loop the animation or stop at last frame
     """
+
     if to_plot not in ['confirmed', 'deaths']:
         raise ValueError("'to_plot' must be in {'confirmed', 'deaths'}")
 
-    # Download data and get dataframe for the given country
-    data = utils.get_json_from_url(config.DATA_URL)
-    df = pd.DataFrame(data[country])
+    if region == "":
+        target = "world.json"
+        url = config.DATA_URL
+    elif country == "Italy":
+        target = "italy.json"
+        url = config.ITALYREGION_URL
+
+    if cached:
+        if os.path.isfile(target):
+            data = json.load(open(target,"r"))
+        else:
+            data = utils.get_json_from_url(url)
+            json.dump(data,open(target,"w"))
+    else:
+        data = utils.get_json_from_url(url)
+
+    if region == "":
+        df = pd.DataFrame(data[country])
+    elif country == "Italy":
+        df = pd.DataFrame(data)
+        df = df.rename(columns=dict(data="date",totale_attualmente_positivi="confirmed",deceduti="deaths",denominazione_regione="region"))
+        if region != "all" and region[0] != "m":
+            df = df[df["region"] == region]
+        else:
+            if region[0] == "m":
+                print("filtering")
```

Comment:
toglierei i prints, o semmai li metterei usando logging (o almeno almeno wrapped in un `if verbose:`, con verbose come parametro della funzione)